src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int RESULT_INDEX = 2;
+    private static final int RECORD_NAME_INDEX = 0;
+    private static final int RECORD_VALUE_INDEX = 1;
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        deleteResultFileIfExist(toFileName);
+        List<String> records = readCsvToList(fromFileName);
+        String report = generateReport(records);
+        writeToFile(report, new File(toFileName));
+    }
+
+    private static String generateReport(List<String> records) {
+        int[] totalSupplyBuyResult = new int[3];
+        calculateSupplyAndBuy(totalSupplyBuyResult, records);
+        totalSupplyBuyResult[RESULT_INDEX] = totalSupplyBuyResult[SUPPLY_INDEX]
+                - totalSupplyBuyResult[BUY_INDEX];
+        String[] resultData = new String[3];
+        resultData[SUPPLY_INDEX] = "supply," + totalSupplyBuyResult[SUPPLY_INDEX];
+        resultData[BUY_INDEX] = "buy," + totalSupplyBuyResult[BUY_INDEX];
+        resultData[RESULT_INDEX] = "result," + totalSupplyBuyResult[RESULT_INDEX];
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < resultData.length; i++) {
+            result.append(resultData[i]).append(System.lineSeparator());
+        }
+        return result.toString();
```

Comment:
it's a bit overcomplicated using array of strings and the builder, just use StringBuilder directly and append everything needed